src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandler.java

```diff
@@ -236,41 +227,58 @@ private void updateProcessingVenueIfRequired(CaseDetails<SscsCaseData> caseDetai
                 }
             }
         }
+    }
+
+    private void updateCaseName(Callback<SscsCaseData> callback, SscsCaseData caseData) {
+        if (!caseAccessManagementFeature) {
+            return;
+        }
 
+        final String caseName = getCaseName(caseData);
+        CaseDetails<SscsCaseData> oldCaseDetails = callback.getCaseDetailsBefore().orElse(null);
+
+        if (oldCaseDetails != null
+            && oldCaseDetails.getCaseData() != null
+            && oldCaseDetails.getCaseData().getCaseAccessManagementFields() != null
+            && oldCaseDetails.getCaseData().getCaseAccessManagementFields().getCaseNameHmctsInternal() != null
+            && oldCaseDetails.getCaseData().getCaseAccessManagementFields().getCaseNameHmctsInternal().equals(caseName)) {
+            return;
+        }
+
+        caseData.getCaseAccessManagementFields().setCaseNames(caseName);
     }
 
-    private void updateCaseNameIfNameUpdated(Callback<SscsCaseData> callback, SscsCaseData caseData) {
-        if (workAllocationFeature) {
-            String caseName = caseData.getAppeal().getAppellant() != null
-                    && caseData.getAppeal().getAppellant().getName() != null
-                    ? caseData.getAppeal().getAppellant().getName().getFullNameNoTitle()
-                    : null;
-
-            CaseDetails<SscsCaseData> oldCaseDetails = callback.getCaseDetailsBefore().orElse(null);
-            if (oldCaseDetails == null
-                    || oldCaseDetails.getCaseData().getWorkAllocationFields().getCaseNameHmctsInternal() == null
-                    || !oldCaseDetails.getCaseData().getWorkAllocationFields().getCaseNameHmctsInternal().equals(caseName)) {
-                caseData.getWorkAllocationFields().setCaseNames(caseName);
-            }
+    private String getCaseName(SscsCaseData caseData) {
+        if (caseData.getAppeal().getAppellant() != null
+            && caseData.getAppeal().getAppellant().getName() != null) {
+            return caseData.getAppeal().getAppellant().getName().getFullNameNoTitle();
```

Comment:
Would be neater if you set the appellant to a local variable before the if

Reply:
Good idea.. changed method to take Appellant object instead.